designs/LicenseFileNameCaseSensitivity.md

```diff
@@ -0,0 +1,63 @@
+
+# Warn on improper license file name case
+
+* Status: In Review
+* Author(s): [David Maas](https://github.com/PathogenDavid)
+* Issue: [9817](https://github.com/NuGet/Home/issues/9817) NU5030 license file existence check should always use a case-sensitive comparison
+
+## Problem Background
+
+Presently when you pack a NuGet package on Windows, it is possible to specify diferent casing for the filename of the `PackageLicenseFile`. For instance, you can pack a file named `License.txt` and specify `<PackageLicenseFile>LICENSE.txt</PackageLicenseFile>` in the NuSpec and the package will pack without errors.
+
+However, this package is malformed because NuGet packages are zip files and zip files use case-sensitive paths internally. As a consequence, the package cannot be uploaded to NuGet.org:
+
+![Screenshot of NuGet.org upload page with error: "The license file 'LICENSE.txt' does not exist in the package."](https://user-images.githubusercontent.com/278957/87859778-464db400-c8fd-11ea-9d19-230c87285e92.png)
+
+If you attempt to put the package in a local feed (or another feed which does not perform this check), you'll get an error:
+
+![Screenshot of Visual Studio error message: "Unknown problem loading the file 'LICENSE.txt'."](https://user-images.githubusercontent.com/278957/87859796-7ac17000-c8fd-11ea-8953-64fc9606c831.png)
+
+## Who are the customers
+
+This primarily affects package authors who may be unknowingly packing malformed packages.
+
+## Goals
+
+* Prevent package authors from making this mistake.
+* Avoid breaking packages which exhibit this problem.
+  * Not all package feeds can or will verify the license file exists. (IE: Local package feeds, Nexus Repository)
+* Fix the behavioral inconsistency between packing on Windows/macOS vs Linux.
+
+## Non-Goals
+
+Changing the behavior of packing on Linux is a non-goal.
+
+## Solution
+
+Introduce a new warning when attempting to pack a package with improper casing on the `PackageLicenseFile`:
```

Comment:
If we accept my proposal to make the warning apply to both embedded licenses as well as embedded icons, should we have one warning code per recourse type, or one warning code and make the message include sufficient information for the customer to understand where to fix the issue?

Reply:
How about we keep the same error code, but slightly change the message?

And do it for both license + icon.

Reply:
If we don't re-use the same error codes like @nkolev92 is suggesting, I would imagine they should be separate error codes since [NU5030](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu5030) and [NU5046](https://docs.microsoft.com/en-us/nuget/reference/errors-and-warnings/nu5046) are already separate.

Reply:
I forgot/didn't know about these existing errors. I very much like the idea of using the same codes and just changing the message when we detect the name with the wrong casing. 👍 

I don't want to add scope creep, but maybe we could also look for the filename in different directories in the nupkg, just in case the user didn't match the nupkg paths?

Reply:
The spec has been updated to use the existing error codes.